### PR TITLE
Implementation of Menter's 1-eq gamma transition model 

### PR DESCRIPTION
**Pull request by BumseokLee**
Hi,

1. 
I would like to begin the draft of pull request for the transition model implementation in Nalu-Wind. 

This is a draft because  there are still some remaining work. However, the basic function of the model is ready and I've completed validation and verification of the model for 2-D airfoils and 3-D cases. The list of the validation (and verification) is as below:

2-D cases
  - ERCOFTAC T3A & T3B flat plate with five different mesh resolution: AIAA Transition Modeling Workshop cases
  - NASA NLF(1)-0416 and NREL S809 with six different mesh resolutions: AIAA Transition Modeling Workshop cases
  - Eppler 387 and NREL S822 at Re=100K, 200K, and 500K: Low Reynolds number cases
  - TU Delft DU00-W-212 at Re=3M to 15M: High Reynolds number case

3-D cases
  - DLR 6:1 Prolate Spheroid
  - NREL Phase VI rotor

All of the validations were performed without any hard-coding. The results were compared against the experimental data as well as available other numerical results, such as NASA's OVERFLOW or FUN3D predictions.

2. 
This work did not begin from scratch but from previous work by Robert Wilson. I’ve attached some related history below.

https://github.com/Exawind/nalu-wind/pull/957
https://github.com/Exawind/nalu-wind/pull/962

He initiated the implementation, but the work wasn’t merged into the main branch due to some issues (pull/957). Additionally, some of his implementation was also reverted (pull/962) due to conflicts with other parts. 

I began my work from the state of pull/962. I fixed some bugs, removed hard coding, and completely re-implemented the computation of the pressure gradient term, which is a key aspect of the current transition model and computed from several different gradients.

3.
Currently, the transition model is only coupled to the k-omega SST model, not with hybrid RANS/LES. This is because there might be interference between the transition model and hybrid RANS/LES. On one hand, the transition model changes the characteristics of the baseline turbulence model and boundary layer. On the other hand, hybrid RANS/LES models (DES/DDES/IDDES) were developed based on fully turbulent flow assumptions with the baseline turbulence model. Therefore, there could be some undesirable interference between them. Once the current work is completed, I will investigate this. 

4. 
I would like to discuss the implementation of the wall boundary condition of the model, which is the most uncertain part for me. The wall boundary condition of the transition model is zero normal flux of the intermittency (gamma). I had a meeting with Phil and asked about this but couldn't fully understand that. I also tried to check the symmetric boundary conditions in Nalu-Wind which should also have zero normal flux across the symmetric plane, it didn't work well. For this reason, I just reused what was implemented before. 
This needs to be definitely double-checked, and any help or suggestions would be extremely appreciated!

5.
We've decided to implement the overset capability of the transition model in ExaWind stack. 
For a typical 3-D wind turbine case, overset exchange of intermittency between different meshes is not important, as the overset boundaries, from the wall to fringe layers, are much larger than the boundary layer thickness. However, in certain cases where there are very small gaps between solid bodies or overlapping surfaces, the oveset exchange between these Nalu-Wind meshes could become important. 

Currently, TIOGA does not distinguish overset exchange between Nalu-Wind / Nalu-Wind and Nalu-Wind / AMR-Wind. Due to this, we decided to implement the overset capability of the transition model in both Nalu-Wind and AMR-wind. The capability is already implemented in Nalu-Wind. For AMR-Wind, we will implement a constant, uniform intermittency field with a value of 1 for transition simulations using the hybrid solvers, which is the freestream conditions of the transition model. Note that AMR-Wind domain is still outside of the boundary layer of the Nalu-Wind meshes and thus it doesn't need to solve the transport equation of the transition model. Therefore, for AMR-Wind, the freestream condition of the transition model will be applied in the computational domain. This implementation will be included in a separate PR of AMR-Wind.( https://github.com/Exawind/amr-wind/pull/1256 )

In summary, a list of the remaining work is as below to complete the implementation of the transition model in Nalu-Wind

- [x] Unit test
- [x] Regression Test
- [x] Confirm the wall boundary condition of the model
- [x] Implementation of the two user options for the freestream turbulence intensity

### CODE REVIEW

File: include/ShearStressTransportEquationSystem.h [hunk 1]
```diff
@@ -92,7 +92,7 @@ class ShearStressTransportEquationSystem : public EquationSystem
 
   const double tkeMinValue_{1.0e-8};
   const double sdrMinValue_{1.0e-8};
-  const double gammaMinValue_{0.0};
+  const double gammaMinValue_{0.02}; // 1/ce2=1/50=0.02
```
Comment (marchdf) [inline include/ShearStressTransportEquationSystem.h:95]: remove comment and prefer signaling in documentation

File: src/SpecificDissipationRateEquationSystem.C [hunk 2]
```diff
@@ -53,6 +53,9 @@
 #include <node_kernels/ScalarGclNodeKernel.h>
 #include <node_kernels/SDRKONodeKernel.h>
 
+// Transition model
```
Comment (marchdf) [inline src/SpecificDissipationRateEquationSystem.C:56]: remove comment

File: src/SpecificDissipationRateEquationSystem.C [hunk 3]
```diff
@@ -269,8 +272,12 @@ SpecificDissipationRateEquationSystem::register_interior_algorithm(
         if (!elementMassAlg)
           nodeAlg.add_kernel<ScalarMassBDFNodeKernel>(realm_.bulk_data(), sdr_);
 
-        if (TurbulenceModel::SST == realm_.solutionOptions_->turbulenceModel_) {
+        if ( TurbulenceModel::SST == realm_.solutionOptions_->turbulenceModel_ &&
```
Comment (marchdf) [inline src/SpecificDissipationRateEquationSystem.C:275]: For clarity I think I would prefer nesting the if conditions. 
If SST
   if gamma active
   else
else if SSTLR

File: src/TurbKineticEnergyEquationSystem.C [hunk 4]
```diff
@@ -62,6 +62,9 @@
 #include <node_kernels/TKERodiNodeKernel.h>
 #include <node_kernels/TKEKONodeKernel.h>
 
+// Transition
```
Comment (marchdf) [inline src/TurbKineticEnergyEquationSystem.C:65]: remove comment

File: src/node_kernels/BLTGammaM2015NodeKernel.C [hunk 5]
```diff
@@ -120,11 +115,18 @@ BLTGammaM2015NodeKernel::execute(
   DblType sijMag = 0.0;
   DblType vortMag = 0.0;
 
-  DblType Ctu1 = 100.;
-  DblType Ctu2 = 1000.;
-  DblType Ctu3 = 1.0;
+  // constants for the source terms
```
Comment (marchdf) [inline src/node_kernels/BLTGammaM2015NodeKernel.C:118]: These are always constant and should be hardcoded? Just checking

File: src/node_kernels/BLTGammaM2015NodeKernel.C [hunk 6]
```diff
@@ -139,31 +141,68 @@ BLTGammaM2015NodeKernel::execute(
   sijMag = stk::math::sqrt(2.0 * sijMag);
   vortMag = stk::math::sqrt(2.0 * vortMag);
 
-  TuL = stk::math::min(
-    81.6496580927726 * stk::math::sqrt(tke) / sdr / (minD + 1.0e-10), 100.0);
+  // Computation of the turbulence intensity
```
Comment (marchdf) [inline src/node_kernels/BLTGammaM2015NodeKernel.C:144]: Can we move this comment and the one below to the documentation? And maybe just make a briefer comment here?

File: src/node_kernels/BLTGammaM2015NodeKernel.C [hunk 7]
```diff
@@ -139,31 +141,68 @@ BLTGammaM2015NodeKernel::execute(
...
+//    caTwo * density * vortMag * fturb * (2.0 * ceTwo * gamint - 1.0);
+//
+//  rhs(0) += (Pgamma - Dgamma) * dVol;
+//  lhs(0, 0) += (DgammaDir - PgammaDir) * dVol;
+
+//========= Jacobian with the Positivity in the implicit operator =============//
+// The original idea by Spalart and Allmaras (1992) for the S-A turbulence model. 
+// The approach adpated by Lee (2021)
+////==========================================================================//
   DblType PgammaDir =
```
Comment (marchdf) [inline src/node_kernels/BLTGammaM2015NodeKernel.C:189]: A bunch of these can be `const`

File: src/node_kernels/SDRSSTBLTM2015NodeKernel.C [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+     sijMag += rateOfStrain * rateOfStrain;
+     vortMag += vortTensor * vortTensor;
+    }
+  }
+  sijMag = stk::math::sqrt(2.0*sijMag);
+  vortMag = stk::math::sqrt(2.0*vortMag);
+
+  Pk = tvisc * sijMag * vortMag; // Pk based on Kato-Launder formulation.
+
+//===============================================================//
```
Comment (marchdf) [inline src/node_kernels/SDRSSTBLTM2015NodeKernel.C:130]: remove these types of comments throughout

File: src/node_kernels/SDRSSTBLTM2015NodeKernel.C [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+    const DblType cEpsOne_two = gammaTwo_ + 1.;
+    const DblType cEpsTwo_two = betaTwo_ / betaStar_ + 1.;
+    const DblType cEpsOneStar_two =
+      cEpsOne_two + (cEpsTwo_two - cEpsOne_two) * (l_t / l_e);
+    gammaTwo_apply = cEpsOneStar_two - 1.;
+  } else {
+    gammaOne_apply = gammaOne_;
+    gammaTwo_apply = gammaTwo_;
+  }
+  //const DblType gamma = fOneBlend * gammaOne_apply + omf1 * gammaTwo_apply;
```
Comment (marchdf) [inline src/node_kernels/SDRSSTBLTM2015NodeKernel.C:168]: remove

**Review by marchdf — CHANGES_REQUESTED** [review]
I just did a pass for esthetics really. We are chatting later this week about other things.

File: include/ShearStressTransportEquationSystem.h (hunk 1 above)
Comment (BumseokLee) [reply include/ShearStressTransportEquationSystem.h]: removed

File: src/SpecificDissipationRateEquationSystem.C (hunk 2 above)
Comment (BumseokLee) [reply src/SpecificDissipationRateEquationSystem.C]: removed

File: src/TurbKineticEnergyEquationSystem.C (hunk 4 above)
Comment (BumseokLee) [reply src/TurbKineticEnergyEquationSystem.C]: removed

File: src/node_kernels/SDRSSTBLTM2015NodeKernel.C (hunk 8 above)
Comment (BumseokLee) [reply src/node_kernels/SDRSSTBLTM2015NodeKernel.C]: removed

File: src/node_kernels/SDRSSTBLTM2015NodeKernel.C (hunk 9 above)
Comment (BumseokLee) [reply src/node_kernels/SDRSSTBLTM2015NodeKernel.C]: removed

File: src/node_kernels/BLTGammaM2015NodeKernel.C (hunk 5 above)
Comment (BumseokLee) [reply src/node_kernels/BLTGammaM2015NodeKernel.C:118]: Discussed.

File: src/node_kernels/BLTGammaM2015NodeKernel.C (hunk 7 above)
Comment (BumseokLee) [reply src/node_kernels/BLTGammaM2015NodeKernel.C]: converted to const

File: unit_tests/kernels/UnitTestKernelUtils.C [hunk 10]
```diff
@@ -158,6 +158,56 @@ struct TrigFieldFunction
                              std::sin(a * pi * z));
   }
 
+  void gamint(const double* coords, double* qField) const //added: transition
```
Comment (marchdf) [inline unit_tests/kernels/UnitTestKernelUtils.C:161]: can you make the name more verbose? We have a lot of "gam/gamma" in the code in general. Maybe `gamma_intermittency"?

File: unit_tests/kernels/UnitTestKernelUtils.C [hunk 11]
```diff
@@ -343,6 +402,12 @@ init_trigonometric_field(
     funcPtr = &TrigFieldFunction::dkdx;
   else if (fieldName == "specific_dissipation_rate")
     funcPtr = &TrigFieldFunction::sdr;
+  else if (fieldName == "gamma_transition") // added: transition
```
Comment (marchdf) [inline unit_tests/kernels/UnitTestKernelUtils.C:405]: remove all `// added: transition" comments. Here and in other files.

File: unit_tests/kernels/UnitTestKernelUtils.h [hunk 12]
```diff
@@ -807,6 +825,7 @@ class SSTKernelHex8Mesh : public LowMachKernelHex8Mesh
       *bulk_, *coordinates_, *density_);
     unit_test_kernel_utils::tke_test_function(*bulk_, *coordinates_, *tke_);
     unit_test_kernel_utils::sdr_test_function(*bulk_, *coordinates_, *sdr_);
+    unit_test_kernel_utils::gamint_test_function(*bulk_, *coordinates_, *gamint_); // added
```
Comment (marchdf) [inline unit_tests/kernels/UnitTestKernelUtils.h:828]: here as well. Comments like `// added` should be removed everywhere.

File: unit_tests/kernels/UnitTestKernelUtils.C (hunk 11 above)
Comment (BumseokLee) [reply unit_tests/kernels/UnitTestKernelUtils.C:405]: Thanks. I forgot to remove these comments. I will do that.

File: src/SpecificDissipationRateEquationSystem.C (hunk 3 above)
Comment (BumseokLee) [reply src/SpecificDissipationRateEquationSystem.C]: I tried this, but it actually made the implementations very lengthy. Let's discuss this during the meeting this afternoon.

**Comment by marchdf** [comment]
Ok we are getting close! Things left to do:
- address the last comments I left
- documentation, mostly in theory/turbulenceModeling
- run the new regression test and unit tests on GPU

File: unit_tests/kernels/UnitTestKernelUtils.C (hunk 10 above)
Comment (BumseokLee) [reply unit_tests/kernels/UnitTestKernelUtils.C:161]: That sounds good to me. I agree that there are so many gam/gamma in the code haha

File: unit_tests/kernels/UnitTestKernelUtils.h [hunk 13]
```diff
@@ -840,6 +860,80 @@ class SSTKernelHex8Mesh : public LowMachKernelHex8Mesh
   sierra::nalu::ScalarFieldType* pecletFactor_{nullptr};
 };
 
+/** Test Fixture for the BLT Gamma Kernels
+ *
+ */
+class BLTGammaM2015KernelHex8Mesh : public LowMachKernelHex8Mesh
```
[the remark this answers is not in the document]
Comment (BumseokLee) [reply unit_tests/kernels/UnitTestKernelUtils.h:866]: Initially, I actually added some small things to SSTKernelHex8Mesh class, but then I created a new class for the transition model since other turbulence models, like the k-ω or k-ε models, have their own classes. However, as you mentioned, the transition model doesn't necessarily need a separate class because it works with the SST model anyway. Additionally, it only requires minor modifications to SSTKernelHex8Mesh due to the similarity between the transition and turbulence models. I will remove the new class and just add those small things to SSTKernelHex8Mesh.

File: unit_tests/kernels/UnitTestKernelUtils.C (hunk 11 above)
Comment (BumseokLee) [reply unit_tests/kernels/UnitTestKernelUtils.C]: done

File: unit_tests/kernels/UnitTestKernelUtils.h (hunk 12 above)
Comment (BumseokLee) [reply unit_tests/kernels/UnitTestKernelUtils.h]: done

File: unit_tests/kernels/UnitTestKernelUtils.C (hunk 10 above)
Comment (BumseokLee) [reply unit_tests/kernels/UnitTestKernelUtils.C]: made changes to gamma_intermittency

File: unit_tests/kernels/UnitTestKernelUtils.h (hunk 13 above)
[the remark this answers is not in the document]
Comment (BumseokLee) [reply unit_tests/kernels/UnitTestKernelUtils.h]: Removed the new class and added some additional lines to SSTKernelHex8Mesh class.

File: src/SpecificDissipationRateEquationSystem.C (hunk 3 above)
Comment (BumseokLee) [reply src/SpecificDissipationRateEquationSystem.C]: Modifications made. Done.

**Comment by marchdf** [comment]
Helping out to make sure the tests are working. 

Unit tests are fine:
```
❯ ./unittestX --gtest_filter="SSTKernelHex8Mesh*"
   Nalu-Wind Version: v2.0.0-89-g8e945c58
   Nalu-Wind GIT Commit SHA: 8e945c58274bddca6c7fff63bbf25c16518d0d7e
   Trilinos Version: 15.1.1                                                                                                                                                                                                                                                        [25/1905]

Note: Google Test filter = SSTKernelHex8Mesh*
[==========] Running 27 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 27 tests from SSTKernelHex8Mesh
[ RUN      ] SSTKernelHex8Mesh.StreletsUpwindComputation
[       OK ] SSTKernelHex8Mesh.StreletsUpwindComputation (39 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_scalar_edge_open_solver_alg
[       OK ] SSTKernelHex8Mesh.NGP_scalar_edge_open_solver_alg (40 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_scalar_open_edge
[       OK ] SSTKernelHex8Mesh.NGP_scalar_open_edge (25 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_tke_sst_node
[       OK ] SSTKernelHex8Mesh.NGP_tke_sst_node (15 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_tke_sst_sust_node
[       OK ] SSTKernelHex8Mesh.NGP_tke_sst_sust_node (9 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_tke_sstlr_node
[       OK ] SSTKernelHex8Mesh.NGP_tke_sstlr_node (12 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_tke_sst_des_node
[       OK ] SSTKernelHex8Mesh.NGP_tke_sst_des_node (12 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_tke_sst_iddes_node
[       OK ] SSTKernelHex8Mesh.NGP_tke_sst_iddes_node (12 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_tke_sst_des_sust_node
[       OK ] SSTKernelHex8Mesh.NGP_tke_sst_des_sust_node (9 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_tke_sst_trans_node
[       OK ] SSTKernelHex8Mesh.NGP_tke_sst_trans_node (12 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_sdr_sst_node
[       OK ] SSTKernelHex8Mesh.NGP_sdr_sst_node (12 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_sdr_sst_sust_node
[       OK ] SSTKernelHex8Mesh.NGP_sdr_sst_sust_node (10 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_sdr_sstlr_node
[       OK ] SSTKernelHex8Mesh.NGP_sdr_sstlr_node (12 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_sdr_sst_des_node
[       OK ] SSTKernelHex8Mesh.NGP_sdr_sst_des_node (12 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_sdr_sst_des_sust_node
[       OK ] SSTKernelHex8Mesh.NGP_sdr_sst_des_sust_node (10 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_sdr_sst_trans_node
[       OK ] SSTKernelHex8Mesh.NGP_sdr_sst_trans_node (12 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_blt_gamma_node
[       OK ] SSTKernelHex8Mesh.NGP_blt_gamma_node (13 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_nodal_grad_edge
[       OK ] SSTKernelHex8Mesh.NGP_nodal_grad_edge (11 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_nodal_grad_elem
[       OK ] SSTKernelHex8Mesh.NGP_nodal_grad_elem (10 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_nodal_grad_elem_shifted
[       OK ] SSTKernelHex8Mesh.NGP_nodal_grad_elem_shifted (9 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_nodal_grad_bndry
[       OK ] SSTKernelHex8Mesh.NGP_nodal_grad_bndry (23 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_nodal_grad_bndry_shifted
[       OK ] SSTKernelHex8Mesh.NGP_nodal_grad_bndry_shifted (19 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_turb_visc_sst_alg
[       OK ] SSTKernelHex8Mesh.NGP_turb_visc_sst_alg (12 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_turb_visc_sstlr_alg
[       OK ] SSTKernelHex8Mesh.NGP_turb_visc_sstlr_alg (9 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_sdr_wall_lowRE
[       OK ] SSTKernelHex8Mesh.NGP_sdr_wall_lowRE (26 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_sdr_wall_func
[       OK ] SSTKernelHex8Mesh.NGP_sdr_wall_func (24 ms)
[ RUN      ] SSTKernelHex8Mesh.NGP_SST_Max_Length_Scale
[       OK ] SSTKernelHex8Mesh.NGP_SST_Max_Length_Scale (11 ms)
[----------] 27 tests from SSTKernelHex8Mesh (422 ms total)

[----------] Global test environment tear-down
[==========] 27 tests from 1 test suite ran. (422 ms total)
[  PASSED  ] 27 tests.
```

**Comment by marchdf** [comment]
And here's the log file for the reg test. It fails with:
```
Cuda Runtime Configuration:
macro  KOKKOS_ENABLE_CUDA      : defined
macro  CUDA_VERSION          = 12050 = version 12.5
Kokkos::Cuda[ 0 ] NVIDIA A100-PCIE-40GB capability 8.0, Total Global Memory: 39.5 G, Shared Memory per Block: 48 K : Selected
Kokkos::Cuda[ 1 ] NVIDIA A100-PCIE-40GB capability 8.0, Total Global Memory: 39.5 G, Shared Memory per Block: 48 K
(CudaInternal::singleton().cuda_device_synchronize_wrapper()) error( cudaErrorIllegalAddress): an illegal memory access was encountered /mnt/vdb/home/mhenryde/exawind/exawind-manager/stage/spack-stage-trilinos-15.1.1-izwc2enrza5c5uyoizre5fgn3ghedsxx/spack-src/packages/kokkos/core/src/Cuda/Kokkos_Cuda_Instance.cpp:153
```
[out.log](https://github.com/user-attachments/files/17135719/out.log)

**Comment by marchdf** [comment]
Ok figured out the problem and pushed the fix. Here's the completed log file: 
[out.log](https://github.com/user-attachments/files/17136043/out.log)

**Comment by BumseokLee** [comment]
Made the CI pass & added the documentation about the transition model.

File: src/node_kernels/TKESSTBLTM2015NodeKernel.C [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+
+  const DblType Ck_BLT = 1.0;
+  const DblType CSEP = 1.0;
+  const DblType Retclim = 1100.0;
+
+  DblType sijMag = 1.0e-16;
+  DblType vortMag = 1.0e-16;
+
+  for (int i = 0; i < nDim_; ++i) {
+    // const int offset = nDim_ * i;
```
Comment (marchdf) [inline src/node_kernels/TKESSTBLTM2015NodeKernel.C:89]: remove comment

File: src/node_kernels/TKESSTBLTM2015NodeKernel.C [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+    for (int j = 0; j < nDim_; ++j) {
+      // const auto dudxij = dudx_.get(node, offset + j);
```
Comment (marchdf) [inline src/node_kernels/TKESSTBLTM2015NodeKernel.C:91]: remove comment

File: src/node_kernels/TKESSTBLTM2015NodeKernel.C [hunk 16]
```diff
@@ -0,0 +1,130 @@
...
+
+  // Pk based on Kato-Launder formulation
+  const DblType Pk = gamint * tvisc * sijMag * vortMag;
+  const DblType Pklim =
+    5.0 * Ck_BLT * stk::math::max(gamint - 0.2, 0.0) * (1.0 - gamint) * Fonlim *
+    stk::math::max(3.0 * CSEP * visc - tvisc, 0.0) * sijMag * vortMag;
+  const DblType Dk =
+    betaStar_ * density * sdr * tke * stk::math::max(gamint, 0.1);
+
+  // Clip production term and prevent Pk from being negative:
```
Comment (marchdf) [inline src/node_kernels/TKESSTBLTM2015NodeKernel.C:118]: remove comment

File: src/node_kernels/TKESSTBLTM2015NodeKernel.C (hunk 14 above)
Comment (BumseokLee) [reply src/node_kernels/TKESSTBLTM2015NodeKernel.C]: done

File: src/node_kernels/TKESSTBLTM2015NodeKernel.C (hunk 15 above)
Comment (BumseokLee) [reply src/node_kernels/TKESSTBLTM2015NodeKernel.C]: done

File: src/node_kernels/TKESSTBLTM2015NodeKernel.C (hunk 16 above)
Comment (BumseokLee) [reply src/node_kernels/TKESSTBLTM2015NodeKernel.C]: done

**Review by marchdf — APPROVED** [review]
